r/R/util.R

```diff
@@ -110,3 +110,9 @@ handle_embedded_nul_error <- function(e) {
   }
   stop(e)
 }
+
+check_tabular <- function(x, format){
+  if(!inherits(x, "ArrowTabular")){
+    stop(paste0("Cannot write to ", format, ". 'x' must be an object of class 'data.frame', 'RecordBatch', or 'Table', not '", class(x)[1], "'"))
```

Comment:
This might be more of a general thing than related to this specific pull request, but perhaps we could use `rlang::abort()` and take advantage of the bullets, e.g. 

```r
abort(c(
  paste0("Cannot write to ", format, ".), 
  x = paste0("'x' must be an object of class 'data.frame', 'RecordBatch', or 'Table', not '", class(x)[1], "'")
))
```

Or if we were willing to use `glue::` 

```r
abort(c(
  glue("Cannot write to { format }."), 
  x = glue("'x' must be an object of class 'data.frame', 'RecordBatch', or 'Table', not '{ class(x)[1] }'")
))
```

Relatedly, we tend to use backticks when naming objects and calls, i.e. 

```r
glue("`x` must be an object") ...
```

Reply:
Great idea, have changed to use `rlang::abort`.